contracts/sc_erc20/ERC20ServiceChain.sol

```diff
@@ -25,8 +25,8 @@ contract ERC20ServiceChain is ERC20, Ownable {
         bridge = _bridge;
     }
 
-    function requestValueTransfer(uint256 _amount, address _to) external {
-        transfer(bridge, _amount);
-        IERC20BridgeReceiver(bridge).onERC20Received(msg.sender, _amount, _to);
+    function requestValueTransfer(uint256 _amount, address _to, uint256 _fee) external {
+        transfer(bridge, _amount.add(_fee));
```

Comment:
What if check revert conditions before transffering?

Reply:
You mean the condition of enough balance?
The enough balance check is already included in that transfer function.

Reply:
I mean the conditions in the `_requestERC20Transfer()`.